tests/devices/i18/test_d7_positioner_a.py

```diff
@@ -0,0 +1,37 @@
+from unittest.mock import ANY
+
+import pytest
+from ophyd_async.core import init_devices
+from ophyd_async.testing import set_mock_value
+
+from dodal.devices.i18.d7positionerA import D7PositionerA, FilterAValues
+
+
+@pytest.fixture
+async def positioner() -> D7PositionerA:
+    async with init_devices(mock=True):
+        positioner = D7PositionerA(prefix="POS:")
+    return positioner
+
+
+async def test_filter_a_setpoint(positioner: D7PositionerA):
+    """
+    Test setting x and y positions on the KBMirror using the ophyd_async mock tools.
+    """
+    # Mock the initial values of the x and y signals
+    set_mock_value(positioner.setpoint, FilterAValues.AL_0_025MM)
+
+    # Call set to update the position
+    await positioner.setpoint.set(FilterAValues.AL_2MM)
+
+    reading = await positioner.read()
+    expected_reading = {
+        "positioner-setpoint": {
+            "value": FilterAValues.AL_2MM,
+            "timestamp": ANY,
+            "alarm_severity": 0,
+        },
+        "positioner-done": {"value": 0.0, "timestamp": ANY, "alarm_severity": 0},
+    }
+
+    assert reading == expected_reading
```

Comment:
nit: making use of ophyd-async's assert_reading this becomes much easier to see what is actually being tested

```suggestion
    await assert_reading(positioner, { "positioner-setpoint": FilterAValues.AL_2MM, "positioner-done": 0.0})
```

Reply:
> FAILED tests/devices/i18/test_d7_positioner_a.py::test_filter_a_setpoint - TypeError: string indices must be integers, not 'str'

> Argument of type "dict[str, str | float]" cannot be assigned to parameter "expected_reading" of type "dict[str, Reading[Unknown]]" in function "assert_reading"


this does not work out of the box 

also having 1 simple and one `new ophyd-async` way of doing things is not Pythonic

(`There should be one-- and preferably only one --obvious way to do it.[[c]](https://en.wikipedia.org/wiki/Zen_of_Python#cite_note-10)`)